grading/__main__.py

```diff
@@ -41,34 +41,55 @@ def init():
 
     config = get_config()
 
-    # Check to see if the config file has a token value
-    if not config['github']['token']:
-        print("A token value is missing. Populating")
-        user = input('GitHub username: ')
-        password = ''
-
-        while not password:
-            password = getpass('Password for {0}: '.format(user))
-
-        note = 'Grading workflow helper'
-        note_url = 'https://github.com/earthlab/grading-workflow-experiments'
-        scopes = ['repo', 'read:user']
-
-        def two_factor():
-            code = ''
-            while not code:
-                # The user could accidentally press Enter before being ready,
-                # let's protect them from doing that.
-                code = input('Enter 2FA code: ')
-            return code
-
-        auth = authorize(user, password, scopes, note, note_url,
-                         two_factor_callback=two_factor)
-
+    # check the token we have is still valid by attempting to login with
+    # the token we have if this fails we need a new one
+    if config.get('github', {}).get('token') is not None:
+        try:
+            # We have to use the GitHub API to find out if our login
+            # credentials actually work, this is why we call `me()`
+            gh = gh3.login(token=config['github']['token'])
```

Comment:
This looks good @betatim . stupid question. you import github as `GH` in this module but here create a `gh` object. I suppose those two things are different caps vs lowercase but just wondering if that will cause confusion in the future. 

Thank you for catching how to properly call `authorize()`. i somehow got stuck at the deprecated note and missed that they told us how to call it properly.

Reply:
re: naming. I think it is already a prime area for mistakes. There is `github3` the library that lets us talk to the GitHub API. There is a `github` in this project which contains custom stuff or higher level things for dealing with GitHub. And finally, the way to use `github3` is to create a GitHub object which I often call `gh`.

I don't know a good way to deal with the name collisions other than establishing a good convention and then sticking with it religiously.